cmd/node/factory/structs.go

```diff
@@ -1314,8 +1320,9 @@ func newShardInterceptorAndResolverContainerFactory(
 	state *State,
 	network *Network,
 	economics *economics.EconomicsData,
-) (process.InterceptorsContainerFactory, dataRetriever.ResolversContainerFactory, error) {
+) (process.InterceptorsContainerFactory, dataRetriever.ResolversContainerFactory, process.BlackListHandler, error) {
 
+	headerBlackList := timecache.NewTimeCache(timeSpanForBadHeaders)
```

Comment:
this creation could be done in the interceptorConteinerFactory. And return it through a public function.

Reply:
interceptorConainerFactory should not have this behavior. It should create the container and that's it. I still think that we should inject as much stuff as possible in order to better unit-test new and existing structs.